### DISCUSSION TITLE
[schema] Composite key

### DISCUSSION THREAD

**Discussion by mdartic** [feathersjs/feathers#3017] [Q&A]
I'm trying to port `feathers-objection` and discover in the `AdapterServiceOptions` that `id` can only be a string

https://github.com/feathersjs/feathers/blob/dove/packages/adapter-commons/src/declarations.ts#L30

How do we manage composite keys ?

For example, in an association table, where the primary key is composed of the two foreign tables.

Actually, I transform the `id` property of service options in a `string | string[]` and rewrite some parts of the `_find` methods like this

```ts
    const idColumns = Array.isArray(this.id) ? this.id : [this.id]
    const fullIdColumns = idColumns.map((idc: string) => `${this.table}.${idc}`)

    // provide default sorting if its not set
    if (!filters.$sort) {
      builder.orderBy(
        fullIdColumns.map((fic: string) => ({
          column: fic,
          order: 'asc',
        })) as ColumnRefOrOrderByDescriptor[],
      )
    }
```

And also for the `count` request.

Do you think this is the "right" way to do it, or maybe there is another way ?

**Comment by sokulski** [comment]
I'm facing a similar challenge. The type declarations for service options don't permit anything besides a string.

Did you manage to get this working?
